### PR TITLE
[Firebase 10] Revamp FIRLogger for Swift use

### PR DESCRIPTION
**Pull request by ncooke3**
See https://github.com/firebase/firebase-ios-sdk/pull/10281#discussion_r982503642 for context!

#no-changelog

### CODE REVIEW

File: Gemfile
Comment (ncooke3) [inline Gemfile:19]: Revert before merging.

File: FirebaseCore/Extension/FIRLogger.h
Comment (ncooke3) [inline FirebaseCore/Extension/FIRLogger.h:179]: FYI: I couldn't find any usage of this API

File: FirebaseCore/Sources/FIRLogger.m
Comment (ncooke3) [inline FirebaseCore/Sources/FIRLogger.m:178]: This implementation could be cleaned up if it just called `FIRLogBasic` but that would use the GULs logger implementation.

The `FIRLog*Swift` functions called into the macro function defined in FIRLogger.m.

Comment (ncooke3) [reply FirebaseCore/Sources/FIRLogger.m:178]: Is there a reason we use the GUL implementation vs: https://github.com/firebase/firebase-ios-sdk/blob/e236acacafffa30de4537c9a7a3028e16f29b9a5/FirebaseCore/Sources/FIRLogger.m#L138-L158

File: FirebaseCore/Sources/FIRLogger.m
Comment (morganchen12) [inline FirebaseCore/Sources/FIRLogger.m:196]: nit: Should an unknown value log as `Error` instead of `Debug`?

**Review by morganchen12 — COMMENTED** [review]
LGTM, but please wait for other approvers before merging.

File: FirebaseCore/Extension/FIRLogger.h
Comment (andrewheard) [reply FirebaseCore/Extension/FIRLogger.h:179]: The NSString/va_list approach isn't as useful in Swift either since string interpolation is concise.

Comment (paulb777) [reply FirebaseCore/Extension/FIRLogger.h:179]: Fine to remove, if also confirmed with an internal global tap.

File: FirebaseCore/Sources/FIRLogger.m
Comment (paulb777) [reply FirebaseCore/Sources/FIRLogger.m:178]: Maybe because FIRLogger was intended to be a wrapper over GULLogger?

**Review by paulb777 — APPROVED** [review]

File: FirebaseCore/Sources/FIRLogger.m
Comment (andrewheard) [reply FirebaseCore/Sources/FIRLogger.m:178]: It looks like the generated macros end up calling `FIRLogBasic` anyway too.

Comment (ncooke3) [reply FirebaseCore/Sources/FIRLogger.m:178]: @paulb777, so you are OK with the current implementation or you think I should remove the above macro and just log through GULs?

File: FirebaseCore/Sources/FIRLogger.m
Comment (ncooke3) [reply FirebaseCore/Sources/FIRLogger.m:196]: Good idea. I will let CI finish and push a fix.

File: Firestore/Swift/Source/Codable/DocumentID.swift
Comment (andrewheard) [inline Firestore/Swift/Source/Codable/DocumentID.swift:127]: nit: I don't feel strongly about it but potentially something like `logIgnoredValueWarning` since that function is for a specific logging case.

**Review by andrewheard — APPROVED** [review]
This LGTM as-is. I'm wondering if it would be best not to worry too much about a cleanup at this point since we might want to look into using [Logger](https://developer.apple.com/documentation/os/logger) when running on iOS 14+ or some other Swifty approach.

File: FirebaseCore/Sources/FIRLogger.m
Comment (ncooke3) [reply FirebaseCore/Sources/FIRLogger.m:178]: > It looks like the generated macros end up calling `FIRLogBasic` anyway too.

@andrewheard, I see it now. I'm going to update my new API's implementation to call directly to `FIRLogBasic`

File: Firestore/Swift/Source/Codable/DocumentID.swift
Comment (ncooke3) [reply Firestore/Swift/Source/Codable/DocumentID.swift:127]: Thanks, I prefer that. Done in 4feb8fc071d23a004f55d9ad619022d071595e33

File: FirebaseCore/Sources/FIRLogger.m
Comment (ncooke3) [reply FirebaseCore/Sources/FIRLogger.m]: Thanks, I prefer that. Done in 4feb8fc071d23a004f55d9ad619022d071595e33

File: FirebaseCore/Extension/FIRLogger.h
Comment (ncooke3) [reply FirebaseCore/Extension/FIRLogger.h:179]: Ehh global TAP is not looking good and I dont want to spend time looking into it now. I'm going to revert some the API breaks.

File: FirebaseCore/Sources/FIRLogger.m
Comment (ncooke3) [reply FirebaseCore/Sources/FIRLogger.m]: I changed the implementation so this is actually obsolete

**Comment by ncooke3** [comment]
[`auth / pod-lib-lint (watchos)` is failing ](https://github.com/firebase/firebase-ios-sdk/actions/runs/3146851345/jobs/5115778295) but I have no idea why. Investigating.....

**Comment by ncooke3** [comment]
> [`auth / pod-lib-lint (watchos)` is failing ](https://github.com/firebase/firebase-ios-sdk/actions/runs/3146851345/jobs/5115778295) but I have no idea why. Investigating.....

Thanks Paul for the tip. The issue is like what was seen in https://github.com/firebase/firebase-ios-sdk/pull/9732. I've fixed the issue locally. I think I will wait for CI to finish to avoid the churn of a restart. 

Assuming no other real failures occur (I expect a flake or two) in the current CI jobs, I will push my fix with a `skip ci` flag after they are done. I have tested my local fix and it looks gtg.

File: Gemfile
Comment (ncooke3) [reply Gemfile]: Done.

**Comment by ncooke3** [comment]
The **latest** rc0 zip is building at https://github.com/firebase/firebase-ios-sdk/actions/runs/3148200619

I merged this before the nightlies started but can diff it with the nightly zip to make sure everything is as expected

**Pull request merged**
